### PR TITLE
[SPARK-28622][SQL][PYTHON] Rename PullOutPythonUDFInJoinCondition to ExtractPythonUDFFromJoinCondition and move to 'Extract Python UDFs'

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR targets to rename `PullOutPythonUDFInJoinCondition` to `ExtractPythonUDFFromJoinCondition` and move to 'Extract Python UDFs' together with other Python UDF related rules.

Currently `PullOutPythonUDFInJoinCondition` rule is alone outside of other 'Extract Python UDFs' rules together.

and the name `ExtractPythonUDFFromJoinCondition` is matched to existing Python UDF extraction rules.

## How was this patch tested?

Existing tests should cover.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -183,10 +183,7 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
     Batch("LocalRelation", fixedPoint,
       ConvertToLocalRelation,
       PropagateEmptyRelation) :+
-    Batch("Extract PythonUDF From JoinCondition", Once,
-      PullOutPythonUDFInJoinCondition) :+
-    // The following batch should be executed after batch "Join Reorder" "LocalRelation" and
-    // "Extract PythonUDF From JoinCondition".
+    // The following batch should be executed after batch "Join Reorder" "LocalRelation".
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: "Join Reorder" and "LocalRelation".

File: sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala [hunk 2]
```diff
@@ -32,6 +32,10 @@ class SparkOptimizer(
   override def defaultBatches: Seq[Batch] = (preOptimizationBatches ++ super.defaultBatches :+
     Batch("Optimize Metadata Only Query", Once, OptimizeMetadataOnlyQuery(catalog)) :+
     Batch("Extract Python UDFs", Once,
+      PullOutPythonUDFInJoinCondition,
```
Comment (gatorsmile) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala]: Rename it to ExtractPythonUDFFromJoinCondition ?

Comment (gatorsmile) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala]: A separate question. Do we have a test case that ensure the local predicate that only contains PythonUDF will be pushed down Join? cc @cloud-fan

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala]: not sure. maybe we can add one in `FilterPushdownSuite`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala (hunk 1 above)
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: From the previous comment, `and` seems to be still missing. :)

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: Argh..

**Comment by gatorsmile** [comment]
LGTM

Merged to master.

**Pull request closed without merging**